base/file.jl

```diff
@@ -1164,3 +1164,94 @@ function chown(path::AbstractString, owner::Integer, group::Integer=-1)
     err < 0 && uv_error("chown($(repr(path)), $owner, $group)", err)
     path
 end
+
+
+# typedef struct uv_statfs_s {
+#     uint64_t f_type;
+#     uint64_t f_bsize;    <- block size
+#     uint64_t f_blocks;   <- total blocks
+#     uint64_t f_bfree;
+#     uint64_t f_bavail;   <- available blocks
+#     uint64_t f_files;
+#     uint64_t f_ffree;
+#     uint64_t f_spare[4];
+# } uv_statfs_t;
+# See also
+# - http://docs.libuv.org/en/v1.x/fs.html#c.uv_fs_statfs (libuv function docs)
+# - http://docs.libuv.org/en/v1.x/fs.html#c.uv_statfs_t (libuv docs of the returned struct)
+struct StatFS
+    ftype::UInt64
+    bsize::UInt64
+    blocks::UInt64
+    bfree::UInt64
+    bavail::UInt64
+    files::UInt64
+    ffree::UInt64
+    fspare1::UInt128
+    fspare2::UInt128
+end
+
+struct DiskStats
+    total::Int
+    available::Int
+end
+
+function DiskStats(path::AbstractString)
+    ispath(path) || throw(ArgumentError("'$path' is not a file or directory. Please provide a valid path."))
```

Comment:
```suggestion
    ispath(path) || throw(ArgumentError("'$path' is not a file or directory."))
```

I personally think this is enough and is consistent with similar error messages, e.g:

```
julia> open("foo")
ERROR: SystemError: opening file "foo": No such file or directory
```

Reply:
thanks, done!